2017/11/29/systemd-primer-from-the-trenches.html.md

```diff
@@ -0,0 +1,514 @@
+reload them without restarting the web service. 
+
+7.
+```systemctl daemon reload```
+
+```chkconfig serviceName --add```
+
+Graceful reloads configuration files on a running service/process. See below for an explanation of “daemon reload”.
+Basically, if you have added in a new service and made many config changes, use daemon-reload.
+
+8.
+```systemctl list-unit-files```
+
+Good flags: --type=service
+
+```ls /etc/rc.d/init.d/```
+```ls /etc/rc.d/rc.local```
+
+Prints from  /usr/lib/systemd/system/ and /etc/systemd/system/
+Slightly different to list-units, rarely used but has any interesting output. You may want to use this in any 
+monitoring scripts you write.
+
+
+9.
+```systemctl list-units```
+
+Good flags -a (--all), -t serviecName, --all
+
+```chkconfig --list``` or ```ntsysv```
+```ls /etc/rc.d/init.d/```
+```ls /etc/rc.d/rc.local```
+```sysv-rc-conf```
+```Initctl list```
+
+Anyone see the problem between the two cmd passes? I prefer to use ‘list-units’ over ‘list-unit-files’. 
+It shows more information and is shorter to type. List all active services using systemctl. 
+Or you could install the sysvinit-utils, which by default are not installed on SysV distro's.
+
+10.
+```systemctl list-sockets```
+
+```lsof```
+```ss -s```
+```cat /proc/net/*```
+
+Sockets of all types can be viewed from one command. Although time to time I will ise lsof or ss depending on the socket
+type I am wanting to look at.
+
+11.
+```systemctl list-timers```
+
+```crontab -e```
+
+systemd offers a way to schedule tasks aka crontab. Im going to go out on a limb here and say they both do the same thing, except
+systemd timers may be more readable by human eyes, are logged to the journal, easier to debug and enable or disable.
+
+Although my caveat; I still use cron jobs in my daily work, because Im familiar with them and emailing is still an issue from a timer.
+
+Tip:
+* set a systemd timer to a calendar, day, month, year to trigger.
+
+
+12.
+```systemctl list-jobs```
+
+Requires further explanation in another blog post.
+
+
+13.
+```systemctl --failed```
+
+Show me failed services. systemctl status will highlight at the top if units have failed, especially after a reboot.
+
+14.
+```systemctl get-default```
+
+```systemctl set-default multi-user.target``` - equivalent to runlevels 2,3 and 4
+
+to list targets run ```systemctl list-units -t target```
+
+```Runlevel```
+```Chkconfig --list```
+```telinit runlevel```
+
+Gets the run level default for the system. Not often used, if rarely, but good know when you start having boot issues or need
+to swap out to a different run level to fix things.
+
+Tip:
+* There are many tables comparing the SysV runlevels to systemd. Get your google on and search.
+* `systemctl set-default graphical.target` will set a graphical user shell. For all those that like a good desktop.
+
+15.
+```systemctl shutdown or reboot```
+
+```shutdown -r now```
```

Comment:
If you show `systemctl shutdown`, should show `shutdown -h now` as the equivalent here.

Reply:
ack

Reply:
added